File: collections/vector_fill_noreserve.c
#include <stdint.h>
#include <stdlib.h>

struct Data {
    uint64_t a;
    uint64_t b;
};

void initData(struct Data* data, size_t x) {
    data->a = x / 2;
    data->b = x / (x % 2 + 1);
}

int main() {
    size_t i;
    struct Data* vec = (struct Data*)malloc(1000000 * sizeof(struct Data));

    for (i = 0; i < 1000000; ++i) {
        initData(&vec[i], i);
    }

    free(vec);
    return 0;
}

File: collections/vector_fill_noreserve.rs
struct Data {
    a: u64,
    b: u64,
}

impl Data {
    fn new(x: usize) -> Data {
        Data {
            a: x as u64 / 2,
            b: x as u64 / (x % 2 + 1) as u64,
        }
    }
}

fn main() {
    let mut vec: Vec<Data> = Vec::new();
    for i in 0..1_000_000 {
        vec.push(Data::new(i));
    }
}